src/fides/api/ops/api/v1/endpoints/consent_request_endpoints.py

```diff
@@ -308,44 +372,39 @@ def set_consent_preferences(
     db: Session = Depends(get_db),
     data: ConsentPreferencesWithVerificationCode,
 ) -> ConsentPreferences:
-    """Verifies the verification code and saves the user's consent preferences if successful."""
+    """Verifies the verification code and saves the user's consent preferences if successful.
+
+    Two workflows are accommodated here:
+    1) Old workflow, where consent preferences are saved with regards to data uses
+    2) New workflow where consent preferences are saved with regards to privacy notices
+    """
     consent_request, provided_identity = _get_consent_request_and_provided_identity(
         db=db,
         consent_request_id=consent_request_id,
         verification_code=data.code,
     )
-    consent_request.preferences = [schema.dict() for schema in data.consent]
-    consent_request.save(db=db)
-
     if not provided_identity.hashed_value:
         raise HTTPException(
             status_code=HTTP_404_NOT_FOUND, detail="Provided identity missing"
         )
 
-    for preference in data.consent:
-        current_preference = Consent.filter(
-            db=db,
-            conditions=(Consent.provided_identity_id == provided_identity.id)
-            & (Consent.data_use == preference.data_use),
-        ).first()
-
-        if current_preference:
-            current_preference.update(db, data=dict(preference))
-        else:
-            preference_dict = dict(preference)
-            preference_dict["provided_identity_id"] = provided_identity.id
-            try:
-                Consent.create(db, data=preference_dict)
-            except IntegrityError as exc:
-                raise HTTPException(
-                    status_code=HTTP_400_BAD_REQUEST, detail=Pii(str(exc))
-                )
+    upserted_consent_preferences: List[Consent] = _save_consent_preferences(
+        provided_identity, db, data.consent
+    )
 
-    consent_preferences: ConsentPreferences = _prepare_consent_preferences(
-        db, provided_identity
+    # Don't re-retrieve all consent preferences from the database: just the preferences
+    # asserted with this request.
+    consent_preferences: ConsentPreferences = ConsentPreferences(
+        consent=upserted_consent_preferences
```

Comment:
This is new, the existing endpoint was re-querying all the consent preferences for the given identity from the database and persisting them (including just the saved ones), but it seems I should only propagate the ones in the request.